src/data/indexHostSingleton.ts

```diff
@@ -36,21 +37,24 @@ export const IndexHostSingleton = (function () {
 
   const key = (config, indexName) => `${config.apiKey}-${indexName}`;
 
-  // describeIndex returns host without the protocol, prepend on return from cache
-  const hostWithProtocol = (host) => `https://${host}`;
-
   return {
     getHostUrl: async function (
       config: PineconeConfiguration,
       indexName: IndexName
     ) {
       const cacheKey = key(config, indexName);
       if (cacheKey in hostUrls) {
-        return hostWithProtocol(hostUrls[cacheKey]);
+        return hostUrls[cacheKey];
       } else {
         const hostUrl = await _describeIndex(config, indexName);
-        hostUrls[cacheKey] = hostUrl;
-        return hostWithProtocol(hostUrl);
+        this._set(config, indexName, hostUrl);
```

Comment:
This felt a bit messier to have duplication on the return rather than just normalizing things before we store it. Also using `this._set()` rather than accessing the cache directly, not sure which would be preferred tbh.